OracleGoldenGate/Dockerfile

```diff
@@ -0,0 +1,85 @@
+# LICENSE CDDL 1.0 + GPL 2.0
+#
+# Copyright (c) 2014-2017 Oracle and/or its affiliates. All rights reserved.
+#
+# ORACLE GOLDENGATE DOCKER PROJECT
+# --------------------------------
+#
+# Dockerfile for Oracle GoldenGate
+#
+# REQUIRED FILES TO BUILD THIS IMAGE
+# ----------------------------------
+#
+# From http://www.oracle.com/technetwork/middleware/goldengate/downloads/index.html
+#  Download the file for "Oracle GoldenGate 12.x.x.x.x for Oracle on Linux x86-64"
+#    - fbo_ggs_Linux_x64_shiphome.zip
+#
+# HOW TO BUILD THIS IMAGE
+# -----------------------
+#
+# Execute the "buildDockerImage.sh" shell script, with the name of the downloaded OGG filename.
+# Example:
+#      $ ./buildDockerImage.sh ~/Downloads/fbo_ggs_Linux_x64_shiphome.zip
+#
+
+ARG         BASE_IMAGE=container-registry.oracle.com/database/instantclient:12.2.0.1
+FROM      ${BASE_IMAGE}
+LABEL       Description="Oracle GoldenGate on Docker" \
+            Maintainer="Stephen Balousek <stephen.balousek@oracle.com>"
+user        root
+
+ARG         OGG_VERSION
+ARG         OGG_EDITION
+ARG         OGG_TARFILE
+
+ENV         JAVA_HOME=/usr/lib/jvm/jre \
+            OGG_HOME=/u01/app/ogg \
+            OGG_DEPLOY_BASE=/u02/ogg \
+            PATH=$PATH:/u01/app/ogg:/u01/app/ogg/bin \
+            OGG_VERSION=${OGG_VERSION} \
+            OGG_EDITION=${OGG_EDITION}
+
+ADD         ${OGG_TARFILE} \
+            runOracleGoldenGate.sh ${OGG_HOME}/
+
+RUN         if [[ -z "${OGG_VERSION}" || -z "${OGG_EDITION}" || -z "${OGG_TARFILE}" ]]; then \
+                echo "----------------------------------------------------------------------------------" && \
+                echo "-- Missing build argument for Dockerfile. See README.md for usage instructions. --" && \
+                echo "----------------------------------------------------------------------------------" && \
+                exit 1; \
+            fi; \
+#
+# Initial setup
+# - Optional proxy setup for Yum
+# - The oracle:oinstall user must use the numeric identifiers 54321:54321
+#
+            [[ -z ${http_proxy} ]] || echo "proxy=${http_proxy}" >> /etc/yum.conf && \
```

Comment:
Without this change to yum.conf, subsequent invocations of yum will fail when executing `docker build` behind a corporate firewall. Since the yum application does not use the `http_proxy` environment variable, this is the only way to communicate proxy details.

One alternative would be to keep this change and *remove* the added lines from yum.conf once all yum operations complete.